lib/gcp/token_validator.go

```diff
@@ -92,5 +93,35 @@ func (id *IDTokenValidator) Validate(ctx context.Context, token string) (*IDToke
 	if err := idToken.Claims(&claims); err != nil {
 		return nil, trace.Wrap(err)
 	}
+
+	if claims.Google.ComputeEngine != (ComputeEngine{}) {
+		return &claims, nil
+	}
+
+	// If the token does not have a `google.compute_engine` claim, it is not a
```

Comment:
Nit: the first two sentences in this comment seem to explain the same thing twice.